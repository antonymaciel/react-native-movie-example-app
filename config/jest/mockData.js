export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

