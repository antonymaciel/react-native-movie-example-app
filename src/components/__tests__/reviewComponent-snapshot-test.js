import React from 'react';
import renderer from 'react-test-renderer';
import { review } from '../../../config/jest/mockData';
import ReviewComponent from '../reviewComponent'


it('renders a ReviewComponent using Snapshots', () => {
  expect(renderer.create(
    <ReviewComponent
        author={review.author}
        url={review.url}
        content={review.content}
    />
  )).toMatchSnapshot();
});
