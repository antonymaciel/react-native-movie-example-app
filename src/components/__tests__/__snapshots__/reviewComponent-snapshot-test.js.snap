// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ReviewComponent using Snapshots 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "marginHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Antony Maciel
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      numberOfLines={5}
      style={
        Object {
          "color": "#696969",
          "fontSize": 15,
          "fontWeight": "500",
          "textDecorationLine": "none",
        }
      }
    >
      Great Movie!, always has amazing content  I love this movie
    </Text>
    <Text
      style={
        Object {
          "color": "#E91E63",
          "textAlign": "right",
          "textDecorationLine": "underline",
        }
      }
    >
      see 
      more
      ...
    </Text>
  </View>
</View>
`;
